smartsim/_core/_install/buildenv.py

```diff
@@ -487,13 +481,31 @@ def _torch_site_path() -> t.Optional[Path]:
                     return torch_path
             return None
 
-        torch_path = _torch_import_path()
-        if not torch_path:
-            torch_path = _torch_site_path()
+        torch_path = _torch_import_path() or _torch_site_path()
         if not torch_path:
             raise SetupError("Could not locate torch cmake path")
         return str(torch_path)
 
+    @staticmethod
+    def find_py_pkg(dist_pkg: str, imp_pkg: t.Optional[str] = None) -> Path:
+        """Find the path to an python import package from a distribution package
+        without checking previously cached modules.
+
+        .. note::
+            Currently this function is only suited to finding namespace dirs or *.py
+            files. It is unable to handle other python importables (*.so, *.zip, etc.)
+        """
+        dist = importlib.metadata.distribution(dist_pkg)
+        imp_pkg = imp_pkg or dist_pkg  # Many distributions ship an import of same name
+        imp_file = os.path.sep.join(imp_pkg.split("."))
+        path = Path(dist.locate_file(imp_file))
+        if path.is_dir():
+            return path
+        path = path.parent / (path.name + ".py")
```

Comment:
f"{path.name}.py"

Reply:
Good catch! best to save the extra mem alloc